src/mondeath.cpp

```diff
@@ -651,7 +651,7 @@ void mdeath::broken( monster &z )
                         const bool uses_mags = !gun.magazine_compatible().empty();
                         if( same_ammo && uses_mags ) {
                             std::vector<item> mags;
-                            int ammo_count = z.ammo[ammo_entry.first];
+                            int ammo_count = rng( 0, z.ammo[ammo_entry.first] );
```

Comment:
I'm not sure how difficult it is, but I'd much rather this was done in JSON than hardcoded.
e.g. Making them spawn with a random amount of ammo.

Reply:
This is not a correct way to solve your problem. The turret should drop exactly as much ammo as it currently has in its inventory. If you want randomized ammo the turret should spawn with a randomized amount of ammo and none of it should magically evaporate.

Reply:
Or the reverse: it had none when it died but all of a sudden it drops 1600 more.